src/plugins/data/public/ui/dataset_selector/dataset_explorer.tsx

```diff
@@ -203,15 +203,23 @@ export const DatasetExplorer = ({
                 >
                   <h3>{current.columnHeader}</h3>
                 </EuiTitle>
-                {current.multiSelect ? (
+                {current.DataStructureCreator ? (
```

Comment:
should we do something similar to index pattern and indices?

instead of having conditional. we just pass the dataset creation component in. in which case it would be  whatever is the default.